tools/bin/record_obfuscator.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+
+import hashlib
+import json
+import sys
+from typing import Any
+
+
+def _generate_hash(value: Any) -> str:
+    return hashlib.sha256(str(value).encode()).hexdigest()
+
+
+def obfuscate(value: Any) -> Any:
+    if isinstance(value, str):
+        obfuscated_value = "string_" + _generate_hash(value)
+    elif isinstance(value, (int, float)):
+        obfuscated_value = "number_" + _generate_hash(value)
+    elif isinstance(value, bool):
+        obfuscated_value = "boolean_" + _generate_hash(value)
+    elif value is None:
+        obfuscated_value = "null_" + _generate_hash(value)
+    elif isinstance(value, list):
+        obfuscated_value = "array_" + _generate_hash(json.dumps(value, sort_keys=True).encode())
+    elif isinstance(value, dict):
+        obfuscated_value = "object_" + _generate_hash(json.dumps(value, sort_keys=True).encode())
+    else:
+        raise ValueError(f"Unsupported data type: {type(value)}")
+
+    return obfuscated_value
+
+
+if __name__ == "__main__":
+    for line in sys.stdin:
+        line = line.strip()
+        try:
+            data = json.loads(line)
+        except Exception as exc:
+            # We don't expect invalid json so if we see it, it will go to stderr
+            print(line, file=sys.stderr)
+        else:
+            if data.get("type") == "RECORD":
+                record_data = data["record"].get("data", {})
+                obfuscated_record = {k: obfuscate(v) for k, v in record_data.items()}
+                data["record"]["data"] = obfuscated_record
+            print(json.dumps(data))
```

Comment:
As `obfuscate` runs on all record I think it's also a cool place to add meta schema generation or other "aggregation" we also perform in live test. We could generate the metaschema from observed record here (like we do in live_test [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/live-tests/src/live_tests/commons/models.py#L318)), write the schema to the container fs and make live test pull it from there. 
Message type counts could also happen here.

Wdyt?